### ISSUE TITLE
Implement a test definitions watcher

### ISSUE THREAD

**Issue by connorshea** [connorshea/vscode-ruby-test-adapter#5]
Instead of needing to reload the test explorer definitions manually, it'd be nice to be able to reload the tests whenever a file in the `spec/` directory is changed.

It may be a good idea to make the watched directory configurable.

**Comment by cristianbica** [comment]
cross-comment from #14

A naive approach would be something like below. The below handles delete / renames also. Would have to check if the file changed is in the test folder. Also a nice one would be to be able to reload only tests from that file.

```diff
diff --git a/src/adapter.ts b/src/adapter.ts
index 8d87d33..96eeef0 100644
--- a/src/adapter.ts
+++ b/src/adapter.ts
@@ -27,6 +27,7 @@ export class RubyAdapter implements TestAdapter {
     this.disposables.push(this.testsEmitter);
     this.disposables.push(this.testStatesEmitter);
     this.disposables.push(this.autorunEmitter);
+    this.disposables.push(this.createWatcher());
   }
 
   async load(): Promise<void> {
@@ -85,4 +86,21 @@ export class RubyAdapter implements TestAdapter {
     let testingFramework: string = (vscode.workspace.getConfiguration('rubyTestExplorer', null).get('testingFramework') as string);
     return testingFramework || 'none';
   }
+
+  private createWatcher(): vscode.FileSystemWatcher {
+    let watcher = vscode.workspace.createFileSystemWatcher("**/*.rb");
+    watcher.onDidChange((file) => {
+      this.log.info(`Reloading test as a file changed: ${file.fsPath}`);
+      this.load();
+    });
+    watcher.onDidCreate((file) => {
+      this.log.info(`Reloading test as a file was created: ${file.fsPath}`);
+      this.load();
+    });
+    watcher.onDidDelete((file) => {
+      this.log.info(`Reloading test as a file was deleted: ${file.fsPath}`);
+      this.load();
+    });
+    return watcher;
+  }
 }
```

**Comment by cristianbica** [comment]
Seen this in N/X Unit .net thing https://github.com/prash-wghats/vscode-nxunit-test-adapter

**Comment by connorshea** [comment]
Implemented in #14 👍
